spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/util/CountingIterator.scala

```diff
@@ -1,13 +1,16 @@
 package com.datastax.spark.connector.util
 
-/** Counts elements fetched form the underlying iterator. */
-class CountingIterator[T](iterator: Iterator[T]) extends Iterator[T] {
+/** Counts elements fetched form the underlying iterator. Limit causes iterator to terminate early */
+class CountingIterator[T](iterator: Iterator[T], limit: Option[Long] = None) extends Iterator[T] {
```

Comment:
I want to draw attention to this change, I added this to minimize over-requesting from C* when we have a known limit.